### PR TITLE
Add multicast support

### LINKED ISSUE #19: Set reuse_port option on create_datagram_endpoint
It would be nice to support reuse of port socket option to allow multiple instances on same machine. I know it's limited to same UID, but still it is nice during development.

### LINKED ISSUE #20: Allow listening to multicast address
It would be nice to configure the gateway device to send to a multicast address.

To make this work properly, the socket needs to be manually constructed and passed to create_datagram_endpoint. This is requires since the standard function will not join multicast addresses, which mean it will not work on setups where switches or wlan access points have igmp snooping enabled.

### PR DESCRIPTION
**Pull request by elupus**
This adds the ability to listen to a multicast target address.

Fixes #19 
Fixes #20

### CODE REVIEW

**Comment by elupus** [comment]
Opening this as draft for now. I need to test this a bit further. But feel free to comment.

File: nibe/connection/nibegw.py [hunk 1]
```diff
@@ -86,10 +88,25 @@
 
         self._set_status(ConnectionStatus.INITIALIZING)
 
+        addrinfo = socket.getaddrinfo(self._listening_ip, self._listening_port)[0]
+        sock = socket.socket(addrinfo[0], socket.SOCK_DGRAM)
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
+        if ip_address(addrinfo[4][0]).is_multicast:
+            group_bin = socket.inet_pton(addrinfo[0], addrinfo[4][0])
+            if addrinfo[0] == socket.AF_INET: # IPv4
+                sock.bind(('', addrinfo[4][1]))
```
[the remark this answers is not in the document]
Comment (yozik04) [reply nibe/connection/nibegw.py:97]: Why does it need to bind to all intefaces?

[the remark this answers is not in the document]
Comment (elupus) [reply nibe/connection/nibegw.py:97]: Which do you suggest it bind to :). Old code also binds to all interfaces. Rather stupid warning. All server code generally bind to all interfaces.

[the remark this answers is not in the document]
Comment (yozik04) [reply nibe/connection/nibegw.py:97]: To self._listening_ip

[the remark this answers is not in the document]
Comment (elupus) [reply nibe/connection/nibegw.py:97]: You cant bind to a multicast address. There is no interface by that ip.

**Pull request merged**